models/ssh_key.go

```diff
@@ -84,7 +84,7 @@ func (key *PublicKey) OmitEmail() string {
 
 // AuthorizedString returns formatted public key string for authorized_keys file.
 func (key *PublicKey) AuthorizedString() string {
-	return fmt.Sprintf(tplPublicKey, fmt.Sprintf(tplCommand, setting.AppPath, setting.CustomConf, key.ID), key.Content)
+	return fmt.Sprintf(tplPublicKey, util.ShellEscape(fmt.Sprintf(tplCommand, util.ShellEscape(setting.AppPath), util.ShellEscape(setting.CustomConf), key.ID)), key.Content)
```

Comment:
I'm having some trouble with this line. I get why the paths need to be escaped, but this looks like it is double escaping things, and can't see why it would need that.

Reply:
because command should be kept together as single value, imho

Reply:
ah yes, I see now. As expected I was just a touch dense. Thanks :)